java/vector/src/main/java/org/apache/arrow/vector/TypeLayout.java

```diff
@@ -225,6 +225,129 @@ public TypeLayout visit(Duration type) {
     return layout;
   }
 
+  /**
+   * Gets the number of {@link BufferLayout}s for the given <code>arrowType</code>.
+   */
+  public static int getTypeBufferCount(final ArrowType arrowType) {
+    return  arrowType.accept(new ArrowTypeVisitor<Integer>() {
+
+      /**
+       * All fixed width vectors have a common number of buffers 2: one validity buffer, plus a data buffer.
+       */
+      static final int FIXED_WIDTH_BUFFER_COUNT = 2;
+
+      /**
+       * All variable width vectors have a common number of buffers 3: a validity buffer,
+       * an offset buffer, and a data buffer.
+       */
+      static final int VARIABLE_WIDTH_BUFFER_COUNT = 3;
+
+      @Override
+      public Integer visit(Int type) {
+        return FIXED_WIDTH_BUFFER_COUNT;
+      }
+
+      @Override
+      public Integer visit(Union type) {
+        switch (type.getMode()) {
+          case Dense:
+            // TODO: validate this
```

Comment:
What is needed to validate this constant?

Reply:
Currently, we do not have a dense union vector in Java code base (one is on-going in https://github.com/apache/arrow/pull/5473). So we need to make sure this is right, once we have a dense union implementation.